letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -272,7 +279,8 @@ def _choose_vhost_from_list(self, target_name):
         elif not vhost.ssl:
             addrs = self._get_proposed_addrs(vhost, "443")
             # TODO: Conflicts is too conservative
-            if not any(vhost.enabled and vhost.conflicts(addrs) for vhost in self.vhosts):
+            if not any(vhost.enabled and vhost.conflicts(addrs) for vhost in self.vhosts)\
+                    and not temp:
```

Comment:
So this actually doesn't quite work. The problem is if `temp` is `True`, the result is cached which will cause problems when an ssl vhost is required in the future (such as when using `run`). This is an easy fix though.